code/modules/mob/living/silicon/robot/component.dm

```diff
@@ -45,30 +45,30 @@
 		destroy()
 
 /datum/robot_component/proc/heal_damage(brute, electronics)
-	if(installed != 1)
+	if(installed != COMPONENT_INSTALLED)
 		// If it's not installed, can't repair it.
-		return 0
+		return FALSE
 
 	brute_damage = max(0, brute_damage - brute)
 	electronics_damage = max(0, electronics_damage - electronics)
 
 /datum/robot_component/proc/is_powered()
-	return (installed == 1) && (brute_damage + electronics_damage < max_damage) && (!energy_consumption || powered)
+	return (installed == COMPONENT_INSTALLED) && (brute_damage + electronics_damage < max_damage) && (!energy_consumption || powered)
 
 
 /datum/robot_component/proc/consume_power()
-	if(toggled == 0)
-		powered = 0
+	if(toggled == FALSE)
+		powered = FALSE
 		return
 	if(owner.cell.charge >= energy_consumption)
 		owner.cell.use(energy_consumption)
-		powered = 1
+		powered = TRUE
 	else
-		powered = 0
+		powered = FALSE
 
 /datum/robot_component/armour
 	name = "armour plating"
-	energy_consumption = 0
+	energy_consumption = FALSE
```

Comment:
Despite all of these only assuming the values 0 or 1 they are actually numbers, not booleans. See https://github.com/vgstation-coders/vgstation13/blob/07a6c73f1a50b0ffee77cd7dd11e6b9a6bb3a932/code/modules/mob/living/silicon/robot/component.dm#L64

Reply:
lovely